src/com/owncloud/android/utils/CopyTmpFileAsyncTask.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    public CopyTmpFileAsyncTask(Activity activity) {
+        mContentResolver = ((FileActivity) activity).getContentResolver();
+        mAccountName = ((FileActivity) activity).getAccount().name;
+        mListener = new WeakReference<OnCopyTmpFileTaskListener>((OnCopyTmpFileTaskListener)activity);
+    }
+
+    @Override
+    protected String doInBackground(Object[] params) {
+        String result = null;
+
+        if (params.length == 3) {
+            Uri uri = (Uri) params[0];
+            String filePath = (String) params[1];
+            mIndex = ((Integer) params[2]).intValue();
+
+            String fullTempPath = FileStorageUtils.getTemporalPath(mAccountName) + filePath;
+            InputStream inputStream = null;
+            FileOutputStream outputStream = null;
+
+            try {
+                inputStream = mContentResolver.openInputStream(uri);
+                File cacheFile = new File(fullTempPath);
+                File tempDir = cacheFile.getParentFile();
+                if (!tempDir.exists()) {
+                    tempDir.mkdirs();
+                }
+                cacheFile.createNewFile();
+                outputStream = new FileOutputStream(fullTempPath);
+                byte[] buffer = new byte[4096];
+
+                int count = 0;
+
+                while ((count = inputStream.read(buffer)) > 0) {
+                    outputStream.write(buffer, 0, count);
+                }
+
+                outputStream.close();
+                inputStream.close();
+
+                result = fullTempPath;
+            } catch (Exception e) {
+                 Log_OC.e(TAG, "Exception ", e);
+                if (inputStream != null) {
+                    try {
+                        inputStream.close();
+                    } catch (Exception e1) {
+                        Log_OC.e(TAG, "Input Stream Exception ", e1);
+                    }
+                }
+
+                if (outputStream != null) {
+                    try {
+                        outputStream.close();
+                    } catch (Exception e1) {
+                        Log_OC.e(TAG, "Output Stream Exception ", e1);
+                    }
+                }
+
+                if (fullTempPath != null) {
+                    File f = new File(fullTempPath);
+                    f.delete();
+                }
+                result =  null;
+            }
+        } else {
+            Log_OC.e(TAG, "Error in parameters number");
+        }
+
+        return result;
+    }
+
+    @Override
+    protected void onPostExecute(String result) {
+
+        OnCopyTmpFileTaskListener listener = mListener.get();
+        if (listener!= null)
+        {
+            listener.OnCopyTmpFileTaskListener(result, mIndex);
+        }
+    }
+
+    /*
+     * Interface to retrieve data from recognition task
+     */
+    public interface OnCopyTmpFileTaskListener{
+
+        void OnCopyTmpFileTaskListener(String result, int index);
```

Comment:
The first letter should not be uppercase in a method; besides, would be better a name not exactly the same as the class, remarking what event just happened; for instance, ```onTmpFileCopied(...)```